acme/acme/challenges.py

```diff
@@ -207,6 +206,74 @@ def response_and_validation(self, account_key, *args, **kwargs):
 
 
 @ChallengeResponse.register
+class DNS01Response(KeyAuthorizationChallengeResponse):
+    """ACME "dns-01" challenge response."""
```

Comment:
nit: Maybe we should remove the quotes around `"dns-01" to be consistent with the other challenge types.